### PR TITLE
"Open At Login" functionality.

### PR DESCRIPTION
**Pull request by ghost**
I wanted to be able to have the application open on startup, so I went ahead and added that functionality.

screenshot_windows

screenshot_macos

I tested it on both Windows 10 and macOS Mojave, and it seems to be working without any errors. 

Edit: Addresses issues #115, #65, and #4

### CODE REVIEW

File: src/menu/settings_menu_template.js [hunk 1]
```diff
@@ -52,6 +53,25 @@ export const settingsMenu = {
...
+      click: (item) => {
+        const openAppAtLoginPref = !settings.get('openAppAtLoginPref');
+        settings.set('openAppAtLoginPref', openAppAtLoginPref);
+        item.checked = openAppAtLoginPref;
+        
+        // Allows Windows & macOS to start the app at login
+        // Note: having the "startup apps" panel open breaks the functionality - Windows bug?
+        const exeName = path.basename(process.execPath);
+        app.setLoginItemSettings({
+        	openAtLogin: item.checked,
```
Comment (ajivoin) [inline src/menu/settings_menu_template.js]: Just for the sake of it being extremely obvious while reading the code, I'm a fan of this line saying:  
```js
openAtLogin: openAppAtLoginPref,
```  
But it doesn't affect functionality so it isn't that big of a deal at the end of the day.

Comment (frankmoskal) [reply src/menu/settings_menu_template.js]: Ah, you're right! Good catch!

**Comment by chrisknepper** [comment]
Hey @frankmoskal , thanks for this! This has definitely been on the to-do list for a while, and it's especially awesome that you've done it without introducing a third-party dependency.

Just a couple of things:

1. Do you know if this can work with Linux?
2. Can you elaborate more on what you mean by it breaking with the "startup apps" panel open on Windows?

File: src/menu/settings_menu_template.js (hunk 1 above)
Comment (ajivoin) [reply src/menu/settings_menu_template.js]: Thanks for making the change, I appreciate it! :)

**Comment by frankmoskal** [comment]
1) According to the [electron docs](https://electronjs.org/docs/api/app#appsetloginitemsettingssettings-macos-windows), it currently only supports Windows & macOS. For Linux, there's [auto-launch](https://www.npmjs.com/package/auto-launch) which could be something to look into for Linux support.

2) Whenever I have the startup apps panel open, any changes I make aren't registered with Windows. It could just be my system, so it would be nice to have some further testing. It works perfectly when the panel isn't open though.

 example panel

**Comment by chrisknepper** [comment]
@frankmoskal ok cool, we may be able to add this to Linux later as the auto-launch docs make it sound pretty simple on that platform.

One thing though, the Electron docs say that since we use auto update, the Windows argument should make the updater exe be pass as well as the regular exe. Could you update the code to do that? Let me know if I'm not making sense here.

**Comment by frankmoskal** [comment]
That should fix it - I'm away from my desktop currently, but I'll test and make sure that works later today.

**Comment by Lainemac** [comment]
I so want this feature.

**Comment by ajivoin** [comment]
Hey @chrisknepper, is there still interest in getting this feature in? I think all we're missing is launching the autoupdater in Windows.

@frankmoskal this is a very late answer, but the location of the updater is in `C:/Users/USER_NAME/AppData/Local/android-messages-desktop-updater` if that helps with implementing this.

**Comment by frankmoskal** [comment]
@ajivoin Thank you! I'll fix this and the conflicts once my exams finish later this week. Would love to get this merged, but I think @chrisknepper was waiting to see if Electron adds native auto-launch support to Linux?

**Pull request closed without merging**